rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -292,10 +250,23 @@ class ReplayRSpace[C, P, A, R, K](store: IStore[C, P, A, K], branch: Branch)(
       }
     }
 
-  def createCheckpoint(): Checkpoint = {
-    val root = store.createCheckpoint()
-    Checkpoint(root, Seq.empty)
-  }
+  private def replaysLessCommRef(replays: ReplayData,
+                                 commRef: COMM): MultisetMultiMap[IOEvent, COMM] =
+    commRef.produces.foldLeft(replays.removeBinding(commRef.consume, commRef)) {
```

Comment:
Why aren't we doing this for `consumes` as well?

Reply:
replays.removeBinding(commRef.consume, commRef)